files/migrations.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/python3
+"""
+Runs migration scripts for this project
+
+Intended to be triggered by the %post rpm scriptlet rather than manually.
+"""
+
+import logging as log
+import shutil
+import sys
+import tempfile
+from importlib.machinery import SourceFileLoader
+from importlib.util import module_from_spec, spec_from_loader
+from pathlib import Path
+
+
+class Version:
+    """
+    A sortable software version
+
+    Reads version strings, and sorts by major.minor.patch[.etc[.etc …]]
+    """
+
+    def __init__(self, version):
+        """
+        Read version numbers into lists of ints, discard anything that comes
+        after the first character that is neither a period or a digit
+        """
+        chunks = []
+        # Opting for minimal code over execution speed
+        for part in version.split("."):
+            digits = ""
+            for char in part:
+                if char.isdigit():
+                    digits += char
+                else:
+                    # Treat RCs, betas etc. as if it were the full release
+                    break
+            chunks.append(int(digits))
+        self.version = tuple(chunks)
+
+    def __str__(self):
+        return ".".join(str(v) for v in self.version)
+
+
+class Migration(Version):
+    """
+    A sortable runnable migration
+
+    Subclass of Version so that we can sort migrations just as easily as version strings
```

Comment:
Is the subclassing necessary? Below I see:
```python3
        migrations = sorted(
            [Migration(p) for p in migrations_dir.glob("*[0-9].py")], key=lambda m: m.version
        )
```
which means we're not sorting the Migration itself, but the `version` member variable. So maybe the output of Version should just be that member variable directly?

Reply:
:100: I agree, subclass was removed. Named the variable `Migration.target` because I find sorting by `migration.version.version` off-putting. Let me know if that works for you